File: backend-express/src/types/enums.ts
export enum HttpStatus {
    OK = 200,
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    NOT_FOUND = 404,
    UNPROCESSABLE_ENTITY = 422,
    INTERNAL_SERVER_ERROR = 500,
}

export enum ContentType {
    multipartFormData = "multipart/form-data",
    applicationJson = "application/json",
}

export enum MimeType {
    JPEG = "image/jpeg",
    PNG = "image/png",
}

export enum CollectionEnum {
    USERS = "users",
    PLACES = "places",
}

export enum Queues {
    EMAILS = "emails",
    AI = "ai",
}

export enum Tasks {
    NEWSLETTER = "newsletter",
    PLACE_EMBEDDING = "place_embedding",
}

File: backend-gin/internal/lib/types_/enums.go
package types_

type ContentType string

const (
	ContentTypeMultipart      ContentType = "multipart/form-data"
	ContentTypeFormUrlencoded ContentType = "application/x-www-form-urlencoded"
	ContentTypeJSON           ContentType = "application/json"
)

func (ct ContentType) IsValid() bool {
	switch ct {
	case ContentTypeMultipart, ContentTypeFormUrlencoded, ContentTypeJSON:
		return true
	default:
		return false
	}
}

type MimeType string

const (
	MimeTypeJPEG MimeType = "image/jpeg"
	MimeTypePNG  MimeType = "image/png"
)

func (mt MimeType) IsValid() bool {
	switch mt {
	case MimeTypeJPEG, MimeTypePNG:
		return true
	default:
		return false
	}
}
